proposals/xxxx-package-manager-performance-testing.md

```diff
@@ -0,0 +1,64 @@
+# Package Manager Performance Testing Conventions
+
+* Proposal: [SE-XXXX](xxx-package-manager-performance-testing.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This is a proposal for adding package manager support for building and running performance tests in release mode.
+
+## Motivation
+
+The package manager supports building and running tests using the `swift test` subtool. The sources are built in debug mode with swift compiler's
+`-enable-testing` flag allowing testable imports in test modules (which gives it internal level access to the imported module). 
+This works well for normal unit tests, however for performance tests we are usually interested in release build test results which takes advantages of compiler 
+optimizations.
+
+We can provide an option to compile and run the tests in release mode with `-enable-testing` flag. However that will disable some of the compiler optimizations, which 
+may be of interest to packages having very performance sensitive code (which critically depends on those optimization).
+
+Since performance tests usually take more time to run than other tests, there is also a requirement to have a way to run performance tests separate from unit tests.
+As a concrete example, package manager's own performance tests are disabled from running on Swift CI.
+
+## Proposed solution
+
+A test module in a Swift package is a directory inside `Tests/` having suffix "Tests". Some properties of test modules are:
+
+* Implicit dependency between any test target and a non-test target that has the same name but without the `Tests` suffix.
+* Executed using `swift test` in debug mode with `-enable-testing` flag.
+
+We propose to extend this convention and consider a directory as "performance test module" which has a suffix "PerformanceTests" with properties:
+
+* Implicit dependency between any test target and a non-test target that has the same name but without the `PerformanceTests` suffix.
+* Executed using `swift test --performance` in release mode without `-enable-testing` flag.
```

Comment:
So `--performance` would ignore the configuration, and force the release configuration?

It is sometimes useful to develop the performance tests in debug mode (for faster build iteration times), and it can even be useful to run them that way. I'm not sure if forcing the configuration is the right answer.

I think it might make sense to decouple the problem of `--enable-testability` from the convention and features for performance tests.